notify-plugin/src/main/java/eu/cloudnetservice/notify/plugin/network/NotifyNetworkHandler.java

```diff
@@ -95,16 +94,21 @@ public void onUpdateOnlineCount(int i) {
 
     }
 
+    /**
+     * Send the notification message to all player they have the permissions
+     * @param key The language key of the message
+     * @param serverInfo The server information's
+     */
     private static void sendNotifyMessage(String key, ServerInfo serverInfo) {
         if (CloudAPI.getInstance().getModuleProperties().contains("notifyService") &&
             CloudAPI.getInstance().getModuleProperties().getBoolean("notifyService")) {
             for (ProxiedPlayer proxiedPlayer : ProxyServer.getInstance().getPlayers()) {
                 if (proxiedPlayer.hasPermission("cloudnet.notify")) {
                     proxiedPlayer.sendMessage(new TextComponent(ChatColor.translateAlternateColorCodes('&',
-                        COMPILE.matcher(CloudAPI.getInstance()
-                                                .getCloudNetwork()
-                                                .getMessages()
-                                                .getString(key)).replaceAll(Matcher.quoteReplacement(serverInfo.getServiceId()
+                        Pattern.compile("%server%", Pattern.LITERAL).matcher(CloudAPI.getInstance()
```

Comment:
If all this is going to do, is to replace a string literal, then we don't need any Pattern.
A plain replace is more efficient here.

If you really want to improve performance, cache these replacements and do them once.